src/v/model/token_bucket.h

```diff
@@ -55,6 +55,20 @@ class throttler {
         return _sem.wait(as, size);
     }
 
+    ss::future<bool> try_throttle(size_t size) {
+        _refresh_timer.cancel();
+        refresh();
+
+        /*
+         * when try_wait succeeds it implies that there are no waiters so there
+         * is no risk in returning without arming the refresh timer.
+         */
+        if (_sem.try_wait(size)) {
+            return ss::make_ready_future<bool>(true);
+        }
+        return ss::make_ready_future<bool>(false);
```

Comment:
co-routine?

Reply:
At first I thought I wanted to leave co-routine interface like in throttle method.
But yeah, there should be simple method without co-routine

Reply:
fixed